server/server.js

```diff
@@ -29,5 +31,44 @@ app.use(`/api/${process.env.SERVER_VERSION}/profile`, profile);
 
 // Run Server
 const host = process.env.HOST || `127.0.0.1`;
-const port = process.env.NODE_PORT || 3001;
+const port = process.env.NODE_PORT || 3009;
 app.listen(port, () => console.log(`Server running on http://${host}:${port}`));
+
+//Establishing Socket.io server
+app.get('/', function (req, res) {
+  res.sendFile(__dirname + '/index.html');
+});
+const nameSpace="";
+    /* @route   http://host:port/nameSpace
+     * @params  {nameSpace}
+     * @desc    opens up a socket.io socket for the specified route
+     */
+io
+    .of(`/${nameSpace}`)
+  
+    .on("connection",function(socket){
+      connections.push(socket);
+      console.log(`Connected to ${nameSpace}, number of connections = ${connections.length}`);
+      socket.emit(`Welcome to ${nameSpace}`);
+    
+    /* @route http://host:port/nameSpace/room
+     * @params{room}
+     * @desc joins or creates a room within the given nameSpace
+     */
+    socket.on("joinRoom",function(room){
+      socket.join(room);
+      io.emit(`New User has joined ${room}`);
+    })
+    /* @route 
+     * @params
+     * @desc disconnects user and returns them to the default room in the nameSpace
+     */
+    socket.on("disconnect",function(data){
+      connections.splice(connections.indexOf(socket),1);
+      console.log(`Connection disconnected, number of connections ${connections.length}`);
+      socket.join("/");
+    })
+    })
+
+
+;
```

Comment:
Have you tested these changes to ensure that it works properly? Also, try moving this code into its own file and including it in here. Like I did with profile and users, if possible

Reply:
yeah it works, don't believe its possible, all tutorials have it running on the server js